src/main/java/org/jenkinsci/main/modules/cli/auth/ssh/UserPropertyImpl.java

```diff
@@ -33,6 +34,13 @@ public boolean has(PublicKey pk) {
         return isAuthorizedKey(signature.asString(pk));
     }
 
+    /**
+     * Gets the users authorized keys
+     */
+    public String getAuthorizedKeys() {
```

Comment:
Whilst I think this is probably OK - its just exposing a list of users public keys, I don't think we should do this as it allows someone to look for compromised keys.

I'll will defer to @Wadeck or @daniel-beck on this (at which point please dismiss my review) - but please do not merge until someone from the security team has reviewed this as this affects all schemes not just the internal user database.

Reply:
🤷‍♂ they're public keys, I don't understand the concern really, how would someone look for them, they either need a malicious plugin which can probably do a lot worse than this,
or they need administer access for the UI, or script console which can already access this.

but if there's more of a concern happy to take a look

Reply:
Public keys themselves **should** not be harmful, however there have been issues in the past [1](https://github.com/g0tmi1k/debian-ssh#the-impact) that the keys where not correctly generated or could have been generated with weak keylength etc, or be otherwise compromised that can be used to launch an attach.  (just like linux does not allow you to view /etc/shadow for example)

my concern is there is a way (I do not know about) to get these keys.  (REST / API seems protected).

Reply:
`User#getProperties` is not `@Exported` so I think this is ok.

Reply:
the field was already public, I've removed the getter as I don't need it for the jcasc change